pwiz_tools/Shared/ProteowizardWrapper/MsDataFileImpl.cs

```diff
@@ -1095,6 +1115,11 @@ private MsDataSpectrum GetSpectrum(Spectrum spectrum, int spectrumIndex)
                         msDataSpectrum.Precursors = ImmutableList.ValueOf(GetMs1Precursors(spectrum));
                     }
 
+                    msDataSpectrum.SourceFilePath = FilePath;
+                    msDataSpectrum.InstrumentInfo = GetInstrumentConfigInfoList().FirstOrDefault();

```

Comment:
The file might have more than one instrument configuration in it so you should get the instrument configuration that is specific for this spectrum.

You can get the "InstrumentConfiguration" object by looking at:
`spectrum.scanList.scans[0].instrumentConfiguration`
but then you would need to turn that into a MsInstrumentConfigInfo.

(There is some example code for looking at the instrument configuration on a spectrum in "pwiz_tools\SeeMS\Dialogs\SpectrumListForm.cs")